parking_lot/parking_lot.py

```diff
@@ -1,8 +1,150 @@
 # -*- coding: utf-8 -*-
-from flask import Flask
+import datetime
 
-app = Flask(__name__)
 
-@app.route("/")
-def hello_world():
-    return "<p>Hello, World!</p>"
+class Parking_lot_error( Exception ):
+    message = 'unhandle error with parking lot'
+
+    def __init__( self, message=None ):
+        if message is None:
+            message = self.message
+        self.data = {
+            'status': 'error',
+            'error': message
+        }
+
+
+class Full_parking_error( Parking_lot_error ):
+    message = 'No free space.'
+
+
+class Cannot_find_car_error( Parking_lot_error ):
+    message = 'Cannot find a car in the location {}.'
+
+    def __init__( self, location=None ):
+        message = self.message.format( location )
+        self.data = {
+            'status': 'error',
+            'error': message
+        }
+
+
+class Car:
+    def __init__(
+            self, car, tariff, location, hourly_fee, daily_fee,
+            start=None, finish=None ):
+        self.car = car
+        self.tariff = tariff
+        self.location = location
+        if start is None:
+            start = datetime.datetime.now()
+        self.start = start
+        self.finish = finish
+
+        self.hourly_fee = hourly_fee
+        self.daily_fee = daily_fee
+
+    def exit( self ):
+        self.finish = datetime.datetime.now()
+
+    @property
+    def fee( self ):
+        if self.is_free_charge:
+            return 0
+
+        if self.is_daily:
+            return self.proportional_diff_time * self.daily_fee
+        if self.is_hourly:
+            return self.proportional_diff_time * self.hourly_fee
+        raise NotImplementedError(
```

Comment:
this exception have another purpose according documentation: https://docs.python.org/3/library/exceptions.html#NotImplementedError

Reply:
partially true, here is been used to indicate where is need to implement new  tariff.

Reply:
done